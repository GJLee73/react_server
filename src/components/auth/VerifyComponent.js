import React, { Component } from 'react';

class VerifyComponent extends Component {
    constructor(props){
        super(props)
        this.state={
            email:''
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleVerify=this.handleVerify.bind(this)
		this.handleKeyPress = this.handleKeyPress.bind(this);        
    }
    handleChange(e){ 
		let nextState = {};
		nextState[e.target.name] = e.target.value;
		this.setState(nextState);
    }
    handleVerify(email){
        var reg=/^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/
        if(!reg.test(email)){
            alert('올바른 이메일을 입력하세요!')
            return 0;
        }
        this.props.checkEmail(email)
    }
    handleKeyPress(e){
		if(e.charCode===13){
			
		    this.handleVerify(this.state.email);
		
        }
    }

    render() { 
        return ( 
        <div>
           
            <div className="container auth">
                <p>인증되지 않은 아이디 입니다. 이메일을 확인해 주세요!!!! 아니면 이메일을 재 전송해 주세요</p>
					<div className="card-content">
						<div className="row">
                        <a className="waves-effect waves-light btn" onClick={()=>{this.props.verify(this.state.email)}}>기존 이메일에 보내기</a>
                       <hr/>
						<div className="input-field col s12">
							<label> 이메일 새로 작성하기</label>
							<input
							name="email"
							type="text"
							className="validate"
							onChange={this.handleChange}
							value={this.state.email}
							onKeyPress={this.handleKeyPress}
                            />
						</div>
                            <a className="waves-effect waves-light btn" onClick={()=>{this.handleVerify(this.state.email)}}>SUBMIT</a>
						</div>
					</div>

				</div>
			</div>
       )
    }
}
 
export default VerifyComponent;