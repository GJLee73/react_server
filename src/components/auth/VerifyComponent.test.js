import React from 'react';
import {shallow} from 'enzyme'
import VerifyComponent from './VerifyComponent'

describe('VerifyComponet test',()=>{
    var handleChange = jest.spyOn(VerifyComponent.prototype,'handleChange')
    var handleKeyPress = jest.spyOn(VerifyComponent.prototype,'handleKeyPress')
    var handleVerify = jest.spyOn(VerifyComponent.prototype,'handleVerify')
    var checkEmail =jest.fn()
    var verify=jest.fn()
    global.alert = jest.fn()

    var wrapper = shallow(<VerifyComponent checkEmail={checkEmail} verify={verify}/>)
    

    describe('function test',()=>{
        it('handleChange test',()=>{
            wrapper.find('input').simulate('change',{target:{value:'jwc2094@naver.com', name:'email'}})
            expect(wrapper.state('email')).toBe('jwc2094@naver.com')
            expect(handleChange).toHaveBeenCalledTimes(1)
        })
        describe('handleKeyPress test',()=>{
            it('should not work',()=>{
                wrapper.find('input').simulate('keypress',{charCode:10})
                expect(handleVerify).toHaveBeenCalledTimes(0)
            })
            it('should work',()=>{
                wrapper.find('input').simulate('keypress',{charCode:13})
                expect(handleVerify).toHaveBeenCalledTimes(1)
                expect(checkEmail.mock.calls.length).toBe(1)
            })
        })
        describe('handleVerify test',()=>{
            it('이메일 양식이 틀릴때',()=>{
                wrapper.setState({email:'enen'})
                wrapper.find('a').at(1).simulate('click')
                expect(global.alert.mock.calls[0][0]).toBe('올바른 이메일을 입력하세요!')
            })
            it('이메일 양식이 맞을때',()=>{
                wrapper.setState({email:'jwc2094@naver.com'})
                wrapper.find('a').at(1).simulate('click')
                expect(global.alert).toHaveBeenCalledTimes(1)
                expect(checkEmail.mock.calls.length).toBe(2)
            })
        })
        it('원래 이메일에 인증메일 다시 보내기',()=>{
            wrapper.find('a').at(0).simulate('click')
            expect(verify.mock.calls.length).toBe(1)
        })
       
    })
    it('snapshot test',()=>{
        expect(wrapper).toMatchSnapshot()
    })
    
})