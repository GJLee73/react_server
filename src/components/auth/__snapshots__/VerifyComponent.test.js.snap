// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyComponet test snapshot test 1`] = `
<div>
  <div
    className="container auth"
  >
    <p>
      인증되지 않은 아이디 입니다. 이메일을 확인해 주세요!!!! 아니면 이메일을 재 전송해 주세요
    </p>
    <div
      className="card-content"
    >
      <div
        className="row"
      >
        <a
          className="waves-effect waves-light btn"
          onClick={[Function]}
        >
          기존 이메일에 보내기
        </a>
        <hr />
        <div
          className="input-field col s12"
        >
          <label>
             이메일 새로 작성하기
          </label>
          <input
            className="validate"
            name="email"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="jwc2094@naver.com"
          />
        </div>
        <a
          className="waves-effect waves-light btn"
          onClick={[Function]}
        >
          SUBMIT
        </a>
      </div>
    </div>
  </div>
</div>
`;
